xds/src/test/java/io/grpc/xds/LookasideChannelLbWithLrsTest.java

```diff
@@ -0,0 +1,227 @@
+  private StreamObserver<DiscoveryResponse> serverResponseWriter;
+  private LookasideChannelLb lookasideChannelLb;
+  private LoadReportClient lrsClient;
+
+  @Before
+  public void setUp() throws Exception {
+    String serverName = InProcessServerBuilder.generateName();
+    AggregatedDiscoveryServiceImplBase serviceImpl = new AggregatedDiscoveryServiceImplBase() {
+      @Override
+      public StreamObserver<DiscoveryRequest> streamAggregatedResources(
+          final StreamObserver<DiscoveryResponse> responseObserver) {
+        serverResponseWriter = responseObserver;
+
+        return new StreamObserver<DiscoveryRequest>() {
+
+          @Override
+          public void onNext(DiscoveryRequest value) {
+            streamRecorder.onNext(value);
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            streamRecorder.onError(t);
+          }
+
+          @Override
+          public void onCompleted() {
+            streamRecorder.onCompleted();
+            responseObserver.onCompleted();
+          }
+        };
+      }
+    };
+
+    cleanupRule.register(
+        InProcessServerBuilder
+            .forName(serverName)
+            .directExecutor()
+            .addService(serviceImpl)
+            .build()
+            .start());
+    channel = cleanupRule.register(
+        InProcessChannelBuilder
+            .forName(serverName)
+            .directExecutor()
+            .build());
+
+    doReturn(channel).when(helper).createResolvingOobChannel(BALANCER_NAME);
+    doReturn(SERVICE_AUTHORITY).when(helper).getAuthority();
+    doReturn(syncContext).when(helper).getSynchronizationContext();
+    doReturn(mock(ChannelLogger.class)).when(helper).getChannelLogger();
+    doReturn(new FakeClock().getScheduledExecutorService()).when(helper)
+        .getScheduledExecutorService();
+
+    LoadBalancerRegistry lbRegistry = new LoadBalancerRegistry();
+    doReturn(localityStore).when(localityStoreFactory).newLocalityStore(helper, lbRegistry);
+    doReturn(loadStatsStore).when(localityStore).getLoadStatsStore();
+    assertThat(streamRecorder.getValues()).isEmpty();
+
+
+    lookasideChannelLb = new LookasideChannelLb(
+        helper, adsStreamCallback, BALANCER_NAME, lrsClientFactory, lbRegistry,
+        localityStoreFactory);
+
+    verify(helper).createResolvingOobChannel(BALANCER_NAME);
+    verify(localityStoreFactory).newLocalityStore(helper, lbRegistry);
+    assertThat(streamRecorder.getValues()).hasSize(1);
+  }
+
+  @After
+  public void tearDown() {
+    lookasideChannelLb.shutdown();
+    verify(lrsClient).stopLoadReporting();
+  }
+
+  /**
+   * Tests load reporting is initiated after receiving the first valid EDS response from the traffic
+   * director, then its operation is independent of load balancing until xDS load balancer is
+   * shutdown.
+   */
+  @Test
+  public void reportLoadAfterReceivingFirstEdsResponseUntilShutdown() throws Exception {
```

Comment:
This can be combined into `LookasideChannelLbTest.java`

Reply:
A mocked `LoadReportClient` is enough, we don't test its real implementation here.